framework/data/Sort.php

```
@@ -295,6 +295,49 @@ public function getAttributeOrders($recalculate = false)
+            }
+            if (!isset($this->attributes[$attribute])) {
+                return false;
+            }
+            $attributes[] = $attribute;
+            if (!$this->enableMultiSort && count($attributes) > 1) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

PHPMD (2.15.0)

[error] Avoid assigning values to variables in if clauses and the like (line '314', column '14'). (undefined)

(IfStatementAssignment)